whois-api/src/test/java/net/ripe/db/whois/api/rdap/RdapServiceTestIntegration.java

```diff
@@ -2282,6 +2296,71 @@ public void lookup_org_entity() {
         assertTnCNotice(notices.get(2), "https://rdap.db.ripe.net/entity/ORG-ONE-TEST");
     }
 
+    @Test
+    public void lookup_redactions() {
+        databaseHelper.addObject("" +
+                "mntner:        ANOTHER-MNT\n" +
+                "descr:         Owner Maintainer\n" +
+                "admin-c:       TP1-TEST\n" +
+                "upd-to:        noreply@ripe.net\n" +
+                "auth:          MD5-PW $1$d9fKeTr2$Si7YudNf4rUGmR71n/cqk/ #test\n" +
+                "mnt-by:        OWNER-MNT\n" +
+                "mbrs-by-ref:   OWNER-MNT\n" +
+                "notify:        test@ripe.net\n" +
+                "created:         2011-07-28T00:35:42Z\n" +
+                "last-modified:   2019-02-28T10:14:46Z\n" +
+                "source:        TEST");
+
+        databaseHelper.addObject("" +
+                "organisation:  ORG-ONE-TEST\n" +
+                "org-name:      Organisation One\n" +
+                "org-type:      LIR\n" +
+                "descr:         Test organisation\n" +
+                "address:       One Org Street\n" +
+                "e-mail:        test@ripe.net\n" +
+                "language:      EN\n" +
+                "admin-c:       TP2-TEST\n" + //has email and notify
+                "tech-c:        TP1-TEST\n" +
+                "tech-c:        TP2-TEST\n" + //has email and notify
+                "mnt-ref:       OWNER-MNT\n" +
+                "mnt-ref:       ANOTHER-MNT\n" +
+                "mbrs-by-ref:   OWNER-MNT, ANOTHER-MNT\n" +
+                "mnt-lower:       ANOTHER-MNT\n" +
+                "mnt-domains:       ANOTHER-MNT\n" +
+                "mnt-routes:       ANOTHER-MNT\n" +
+                "mnt-by:        ANOTHER-MNT\n" +
+                "created:         2011-07-28T00:35:42Z\n" +
+                "last-modified:   2019-02-28T10:14:46Z\n" +
+                "source:        TEST");
+
+        final Entity entity = createResource("entity/ORG-ONE-TEST")
+                .request(MediaType.APPLICATION_JSON_TYPE)
+                .get(Entity.class);
+
+        assertThat(entity.getRedacted().size(), is(9));
+
+        assertRedaction(entity.getRedacted().get(0), "Updates notification e-mail information", "Personal data", "$.entities[?(@.roles=='registrant')].vcardArray[1][?(@[0]=='notify')]");
```

Comment:
Notify is optional attribute , there can be few registrant with no notify attribute and some might, so we are telling user that we removed notify from all entities which is wrong, I think we need to mention the index here

Reply:
email is optional as well and it is the example that the RFC is using.
I think that as far as we notify the user about the changes that we do it is not necessary to exactly point in what objects we did the change.

Reply:
I do not think RFC examples are made specifically thinking about that. Although email is mandatory for few object types.  prePath should point to one or more elements that we redacted from the original json. Since we include key value in original response, we should not give false impression for removing a field that is not there